core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java

```diff
@@ -6122,6 +6122,54 @@ private Sql checkSubQuery(String sql) {
     checkPlanning(FilterJoinRule.FILTER_ON_JOIN, query);
   }
 
+  @Test public void testWindowOnSortedInput() {
+    // Create a customized test with RelCollation trait in the test cluster.
+    Tester tester = new TesterImpl(getDiffRepos(), true, true, false, false,
+        true, null, null) {
+      @Override public RelOptPlanner createPlanner() {
+        return new MockRelOptPlanner(Contexts.empty()) {
+          @Override public List<RelTraitDef> getRelTraitDefs() {
+            return ImmutableList.of(RelCollationTraitDef.INSTANCE);
+          }
+          @Override public RelTraitSet emptyTraitSet() {
+            return RelTraitSet.createEmpty().plus(
+                RelCollationTraitDef.INSTANCE.getDefault());
+          }
+        };
+      }
+    };
+
+    final HepProgram preProgram = new HepProgramBuilder()
+        .addRuleInstance(SortProjectTransposeRule.INSTANCE)
+        .build();
+    final HepProgram program = HepProgram.builder()
+        .addRuleInstance(ProjectToWindowRule.PROJECT)
+        .build();
```

Comment:
From the tests you gave, i couldn't figure out what you are fixing, because both the input and output nodes collation are the same.

Reply:
Nope, not the same. For output project collation is [0, 1]. And before the fix it was copied to LogicalWindow.

Reply:
Oh, you mean the collation field indices ? That make sense.

Reply:
Yes, project reorders fields so collation indices are changed. Actually there was a bug with output, fixed now and provided checks for both input and output collations.